### ISSUE TITLE
Issue of INVALID_TOKEN after one hour in samples/ZohoCRM.gs (Refer #197)

### ISSUE THREAD

**Issue by pankajk79** [gsuitedevs/apps-script-oauth2#198]
Added 
// Set prompt to "consent" to ensure a refresh token is retrieved.
.setParam('prompt', 'consent')

Still getting same error after one hour

Please look at this, urgently needed.

**Comment by erickoledadevrel** [comment]
Did you reset the service and reauthorize it? You can call `Logger.log(service.getToken());` to see if you have a refresh token.

**Comment by pankajk79** [comment]
Thanks for responses.

I have reset the service and reauthorize.

authorizationUrl containing both access_type=offline and prompt=consent at the end.

Added Logger.log(service.getToken()); 

Both access_token and refresh_token returns with other values.

After one hour, message logged
[19-08-24 11:47:48:741 IST] {
  "code": "INVALID_TOKEN",
  "details": {},
  "message": "invalid oauth token",
  "status": "error"
}

Again after one hour Logger.log(service.getToken());  returns same values as that was one hour before.

What does that mean? 

Ideally  a new access_token should be returned. Right?

**Comment by pankajk79** [comment]
Please look at this issue.

**Comment by pankajk79** [comment]
I got a solution for this.
I have copied code from "dist" folder to my code.
Changed 
var expiresIn = token.expires_in || token.expires ;

to 

var expiresIn = token.expires_in_sec || token.expires_in || token.expires ;

at line 720 because Zoho CRM return access token expiration time as expires_in_sec=3600

After this generated token again.

It is now two days, token is not expired.

Thanks for your suggest to include .setParam('prompt', 'consent') while creating service.
